source/validate_instruction.cpp

```diff
@@ -340,9 +341,44 @@ spv_result_t RegisterDecorations(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+// Parses OpExtension instruction and registers extension.
+spv_result_t RegisterExtension(ValidationState_t& _,
+                               const spv_parsed_instruction_t* inst) {
+  assert(inst->opcode == SpvOpExtension);
+  assert(inst->num_operands == 1);
+
+  const auto& operand = inst->operands[0];
+  assert(operand.type == SPV_OPERAND_TYPE_LITERAL_STRING);
+  assert(inst->num_words > operand.offset);
+
+  const size_t max_length =
```

Comment:
The binary parser already does the check for the nul terminator.   So this is a redundant sequence of code to compute max bounds and check again.